### PR TITLE
feat: prof modal

### PR DESCRIPTION
**Pull request by filippo-fonseca**
## Pre-flight checklist

- [x] I have searched for potential duplicate pull requests.
- [ ] **If this is a code change**: I have written unit tests and/or conducted manual tests to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Summary

This PR fixes/implements the following **bugs/features**

- Implements the professor modal state
- Provides a beautiful chart to view how the professor's rating has changed over time (avg.)
- Merge conflicts with master branch back-tracked, updated, and fixed
- Null ratings and problems with data parsing have all been addressed by virtue of a `CustomCart` component

## Test plan (required)

Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

Open any course on CT and click on the professor's name in the overview section!

Fix https://github.com/coursetable/coursetable/issues/1687

### CODE REVIEW

File: frontend/src/pages/About.tsx [hunk 1]
```diff
@@ -13,6 +13,7 @@ import dl from '../images/headshots/deyuan-li.jpg';
 import df from '../images/headshots/dylan-fernandez-de-lara.jpg';
 import eb from '../images/headshots/erik-boesen.jpg';
 import eh from '../images/headshots/evan-hu.jpg';
+import ff from '../images/headshots/filippo-fonseca.jpg';
```
Comment (Josh-Cena) [inline frontend/src/pages/About.tsx:16]: Could you move these changes out to a separate PR (will merge after merging this one)?

Comment (filippo-fonseca) [reply frontend/src/pages/About.tsx]: done!

**Comment by Josh-Cena** [comment]
Cross-posting my feedback:

1. The graph should show the course code somewhere, either on the x-axis or the hover text of the nodes (preferably latter since x-axis being year makes sense)
2. Preferably each node should be clickable and take you to the evals of that course (not required)
3. We should move the "prof" column on the course modal here, such that the "prof" column of the course modal now just shows a message saying: "we've moved, use one of the links below to see their ratings: [list of professor names]", and the prof modal also has a two-column layout, one for the graphs, one for the list of classes with the color-coded overall/workload stats (no need for prof rating here)
4. My idea is still that all professor modals for the same prof should look exactly identical. That is, it doesn't matter if you opened Glenn's modal from CPSC 223 or CPSC 470; they should show you exactly the same text, without the course's label.
5. The prof modal should be part of the URL. For example, if you are on `https://coursetable.com/catalog?course-modal=202501-20168`, then clicking on "James Glenn" should replace the URL with `https://coursetable.com/catalog?prof-modal=[professor_id]`. This way, you can use regular `<Link>` links for opening up the modals.

**Comment by neilsong** [comment]
Also, you should make the tooltip hover appear even when not directly on the point/node. Here's an example:

https://www.chartjs.org/docs/latest/samples/tooltip/interactions.html

**Comment by filippo-fonseca** [comment]
Update: Should be done by either today or tomorrow afer what we talked about yesterday

**Comment by bearsyankees** [comment]
amazing @filippo-fonseca
